lib/fika/parser_helper.ex

```diff
@@ -135,6 +145,10 @@ defmodule Fika.ParserHelper do
     end
   end
 
+  def do_to_ast({[{:identifier, line, value}], line}, :atom) when value in [true, false] do
+    {:boolean, line, value}
+  end
```

Comment:
Why do we do this?

Reply:
I suggested this so that our intermediate representation doesn't represent `:true` and `:false` differently.
Altough at the Erlang level they will be converted back to atoms, I think that some problems could arise in the type checker if we didn't have this clause

Reply:
Atoms - `:true` and `:false` in Fika are of the types `:true` and `:false`. Shouldn't they be represented differently from boolean true/false in our AST?

Reply:
I disagree, because  at the BEAM level they will both be represented as atoms. Having them be different types would also make interop with Erlang and other BEAM languages much more difficult. I think that they should be booleans because of this. 

We could also go as far as disallowing `:true` and `:false` (and maybe `:nil` in the future?) as atoms altogether. This would also resolve this problem to me. In this case, the fix is not pertinent to this PR.

Reply:
[This line](https://github.com/fika-lang/fika/pull/16/files#diff-cb8f4714e7283bea2cc1d4f70a924bddR99) shows that they are indeed atoms at the lowest level